Tiltfile

```diff
@@ -137,87 +145,36 @@ helm_resource(
-    configmap_create(
-        "envoy-config",
-        from_file="./local/path/envoy/.envoy.yaml",
-        watch=True,
-    )
-    configmap_create(
-        "allowed-services",
-        from_file="./local/path/envoy/.allowed-services.lua",
-        watch=True,
-    )
-    configmap_create(
-        "gateway-endpoints",
-        from_file="./local/path/envoy/.gateway-endpoints.yaml",
-        watch=True,
-    )
-    configmap_create(
-        "ratelimit-config",
-        from_file="./local/path/envoy/.ratelimit.yaml",
-        watch=True,
-    )
-
-    # 1. Load the Kubernetes YAML for the envoy-proxy service
-    k8s_yaml("./local/kubernetes/envoy-proxy.yaml")
-    k8s_resource(
-        "envoy-proxy",
-        labels=["envoy_auth"],
-        # By default the Envoy Proxy container will bind to 127.0.0.1.
-        # Adding 0.0.0.0 allows it to be accessible from any IP address.
-        port_forwards=["0.0.0.0:3070:3070"],
-    )
-
-    # 2. Build the External Auth Server image from envoy/auth_server/Dockerfile
-    docker_build(
-        "ext-authz",
-        context="./envoy/auth_server",
-        dockerfile="./envoy/auth_server/Dockerfile",
-        live_update=[
-            sync("./envoy/auth_server", "/app"),
-        ],
-    )
-    # Load the Kubernetes YAML for the External Auth Server
-    k8s_yaml("./local/kubernetes/envoy-ext-authz.yaml")
-    k8s_resource(
-        "ext-authz",
-        labels=["envoy_auth"],
-        port_forwards=["10003:10003"],
-        resource_deps=["path-auth-data-server", "path-config-updater"],
-        trigger_mode=TRIGGER_MODE_AUTO,
-    )
-
-    # 3. Load the Kubernetes YAML for the path-auth-data-server service
-    k8s_yaml("./local/kubernetes/envoy-pads.yaml")
-    k8s_resource(
-        "path-auth-data-server",
-        labels=["envoy_auth"],
-    )
-
-    # 4. Load the Kubernetes YAML for the ratelimit service
-    k8s_yaml("./local/kubernetes/envoy-ratelimit.yaml")
-    k8s_resource(
-        "ratelimit",
-        labels=["envoy_auth"],
-        resource_deps=["redis"],
-    )
-
-    # 5. Load the Kubernetes YAML for the redis service
-    k8s_yaml("./local/kubernetes/envoy-redis.yaml")
-    k8s_resource(
-        "redis",
-        labels=["envoy_auth"],
+    # The folder containing GUARD's local configuration files.
+    GUARD_LOCAL_DIR = LOCAL_DIR + "/guard"
+    # The values file for GUARD's Helm chart.
+    GUARD_LOCAL_VALUES_FILE = GUARD_LOCAL_DIR + "/.values.yaml"
+
+    # New resources created from Helm Charts
+    helm_resource(
+        "guard",
+        chart_prefix + "guard",
+        namespace=NAMESPACE,
+        labels=["guard"],
+        flags=[
+            "--values=" + GUARD_LOCAL_VALUES_FILE,
+        ]
     )
+    # Patch the Envoy Gateway LoadBalancer resource to ensure 
+    # it is reachable from outside the cluster at "localhost:3070".
+    #
+    # For more context, see `./local/scripts/patch_envoy_gateway.sh`.
```

Comment:
Please add a TODO: this needs to be done by exposing a port on GUARD.

Reply:
> Please add a TODO: this needs to be done by exposing a port on GUARD.

Please read the comments in `./local/scripts/patch_envoy_gateway.sh` as they explain why exposing a port of GUARD does not work in Tilt and why this exists.